### ISSUE TITLE
Out of stock subproduct may cause min price wrong

### ISSUE THREAD

**Issue by VictorSnow** [algolia/algoliasearch-magento-2#442]
**Do you want to request a *feature* or report a *bug*?**
bug

**Bug: What is the current behavior?**
min price is using out of stock product price
has been set out of stock option to no in algolia settings and inventory settings
**Bug: What is the expected behavior?**
use min price of in stock products

**What is the version of Magento and of Algolia extension you are using? Always use the latest version of the extension one before opening a bug issue.**
Magento 2.2.3

**Comment by JanPetr** [comment]
Thank you for the report @VictorSnow, I'll take a look on the issue.

**Comment by peterjaap** [comment]
Coincidentally I just ran into the solution for this.

We need to add this piece of code;

```
            $isInStock = (bool) $this->stockRegistry->getStockItem($subProduct->getId())->getIsInStock();
            if ($isInStock === false && $this->configHelper->indexOutOfStockOptions($storeId) === false) {
                continue;
            }
```

In the foreach loop here; https://github.com/algolia/algoliasearch-magento-2/blob/e8500a9f489bf9573bc18cac37a3e647adb02d69/Helper/Entity/Product/PriceManager.php#L386

(of course also inject `Magento\CatalogInventory\Api\StockRegistryInterface` into `$stockRegistry`)

**Comment by JanPetr** [comment]
Hello, the issue was fixed in #462 and will be part of the next release.
